l10n_ir_base/models/fa_models.py

```diff
@@ -0,0 +1,95 @@
+from odoo import models,api
+
+import logging
+import pytz
+import babel.dates
+from odoo.tools.misc import DEFAULT_SERVER_DATETIME_FORMAT, DEFAULT_SERVER_DATE_FORMAT, get_lang
+from odoo.osv import expression
+
+import jdatetime as jd
+
+_logger = logging.getLogger(__name__)
+
+def display_format_changer(df):
+        return {
+        'hh:00 dd MMM': '%d %b %H',
+        'dd MMM yyyy': '%d %b %Y',
+        "'W'w YYYY": '%A %Y',
+        'MMMM yyyy': '%b %Y',
+        'QQQ yyyy': '%b %Y',
+        'yyyy': '%Y',
+    }.get(df, '%d %b %Y') 
+    
+
+class BaseModelExtend(models.AbstractModel):
+    _inherit = 'base'
+    @api.model
+    def _read_group_format_result(self, data, annotated_groupbys, groupby, domain):
+        sections = []
+        for gb in annotated_groupbys:
+            ftype = gb['type']
+            value = data[gb['groupby']]
+
+            # full domain for this groupby spec
+            d = None
+            if value:
+                if ftype == 'many2one':
+                    value = value[0]
+                elif ftype in ('date', 'datetime'):
+                    locale = get_lang(self.env).code
+                    fmt = DEFAULT_SERVER_DATETIME_FORMAT if ftype == 'datetime' else DEFAULT_SERVER_DATE_FORMAT
+                    tzinfo = None
+                    range_start = value
+                    range_end = value + gb['interval']
+                    # value from postgres is in local tz (so range is
+                    # considered in local tz e.g. "day" is [00:00, 00:00[
+                    # local rather than UTC which could be [11:00, 11:00]
+                    # local) but domain and raw value should be in UTC
+                    if gb['tz_convert']:
```

Comment:
what about using fields.Datetime methods to use the tz transformation of the user

Reply:
I copied this direct part of the original code and just added the display_format_changer function

Reply:
https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_journal_dashboard.py

Reply:
https://github.com/odoo/odoo/blob/14.0/odoo/models.py
line 2181